_site/posts/2022-09-16_arrow-and-rpy2/index.html

```diff
@@ -0,0 +1,1102 @@
+<pre><code>pyarrow.Table
+Gender: string
+Nationality: string
+n: int64
+----
+Gender: [["Unknown","Male","Female","Male","Female",...,"Both","Female","Female","Female",null]]
+Nationality: [["Unknown","British","British","Australian","Australian",...,"Australian/British","British/American","South African","Polish","Australian"]]
+n: [[12832,6420,3346,2537,1687,...,1,1,1,1,1]]</code></pre>
+</div>
+</div>
+<p>Just like that, we’ve handed over the Arrow Table from R back to Python. Again, it helps to remember that <code>gender2</code> is an R object and <code>gender3</code> is a Python object, but both of them point to the same underlying Arrow Table.</p>
+<p>In any case, now that we have <code>gender3</code> on the Python side, we can use the <code>to_pandas()</code> method from <code>pyarrow.Table</code> to convert it to a pandas data frame:</p>
+<div class="cell" data-execution_count="15">
+<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>gender4 <span class="op">=</span> pyarrow.Table.to_pandas(gender3)</span>
+<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>gender4</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
+<div class="cell-output cell-output-display" data-execution_count="15">
+
+<div>
+
+<table class="dataframe table table-sm table-striped">
+  <thead>
+    <tr>
+      <th></th>
+      <th>Gender</th>
+      <th>Nationality</th>
+      <th>n</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>0</th>
+      <td>Unknown</td>
+      <td>Unknown</td>
+      <td>12832</td>
+    </tr>
+    <tr>
+      <th>1</th>
+      <td>Male</td>
+      <td>British</td>
+      <td>6420</td>
+    </tr>
+    <tr>
+      <th>2</th>
+      <td>Female</td>
+      <td>British</td>
+      <td>3346</td>
+    </tr>
+    <tr>
+      <th>3</th>
+      <td>Male</td>
+      <td>Australian</td>
+      <td>2537</td>
+    </tr>
+    <tr>
+      <th>4</th>
+      <td>Female</td>
+      <td>Australian</td>
+      <td>1687</td>
+    </tr>
+    <tr>
+      <th>...</th>
+      <td>...</td>
+      <td>...</td>
+      <td>...</td>
+    </tr>
+    <tr>
+      <th>58</th>
+      <td>Both</td>
+      <td>Australian/British</td>
+      <td>1</td>
+    </tr>
+    <tr>
+      <th>59</th>
+      <td>Female</td>
+      <td>British/American</td>
+      <td>1</td>
+    </tr>
+    <tr>
+      <th>60</th>
+      <td>Female</td>
+      <td>South African</td>
+      <td>1</td>
+    </tr>
+    <tr>
+      <th>61</th>
+      <td>Female</td>
+      <td>Polish</td>
+      <td>1</td>
+    </tr>
+    <tr>
+      <th>62</th>
+      <td>None</td>
+      <td>Australian</td>
+      <td>1</td>
+    </tr>
+  </tbody>
+</table>
+<p>63 rows × 3 columns</p>
+</div>
+</div>
+</div>
+<p>And with that our transition home is complete!</p>
```

Comment:
Could there be a longer conclusion, to tie up the journey and restate its value? While this is an informative article, it's presented with such a narrative voice (which is fun to read!) that this feels abrupt.

Reply:
This is a very bad habit of mine, that has plagued me my whole career. I am terrible at writing conclusions. I sucked at it as an academic and I'm amused to see that it's still a problem for me now. You are right though - this needs a bit more. I'll see what I can do